### DISCUSSION TITLE
Firewall ports to be opened as part of Keda Deployment

### DISCUSSION THREAD

**Discussion by nidiculageorge** [kedacore/keda-docs#389] [General]
Hi Team,

We need to identify which all ports and URLS need to be allowed in our azure firewall for Keda to communicate outside.

Could you please share the details of firewall URLS and Ports to be whitelisted in our azure firewall

**Comment by tomkerkhove** [comment]
@ahmelsayed @zroubalik Can you help on this?

I'll merge this in the docs afterwards

**Reply by ahmelsayed** [reply]
_<-- originally posted on slack -->_

How is it failing? are you getting a particular error?
egress from keda will depend on the triggers you have defined as it'll try to connect to each one you define. Some like SQL, redis, rabbitmq, etc specify their ports in the ScaledObject itself.

**Reply by nidiculageorge** [reply]
Hi @ahmelsayed 

{"level":"info","ts":1614313669.8391643,"logger":"cmd","msg":"Go Version: go1.13.3"}
{"level":"info","ts":1614313669.8396757,"logger":"cmd","msg":"Go OS/Arch: linux/amd64"}
{"level":"info","ts":1614313669.8396797,"logger":"cmd","msg":"Version of operator-sdk: v0.11.0"}
{"level":"info","ts":1614313669.8396823,"logger":"cmd","msg":"Keda Commit: "}
{"level":"info","ts":1614313669.839852,"logger":"leader","msg":"Trying to become the leader."}
{"level":"error","ts":1614313669.8433106,"logger":"cmd","msg":"","error":"Get https://10.1.0.1:443/api?timeout=32s: EOF","stacktrace":"github.com/go-logr/zapr.(*zapLogger).Error\n\t/go/pkg/mod/github.com/go-logr/zapr@v0.1.1/zapr.go:128\nmain.main\n\tkeda/cmd/manager/main.go:89\nruntime.main\n\t/usr/local/go/src/runtime/proc.go:203"}

When checked the logs for Keda we are getting the above error in Azure AKS .There is no triggers enabled as its just a keda pod thats running

**Reply by ahmelsayed** [reply]
I think that's the cluster's api-server. That's the only endpoint keda connects to on start up. I'm not very familiar with azure firewall, but I'd imagine there must be a way to allow pods to talk to the api server. Any operator would need that.

**Comment by nidiculageorge** [comment]
@ahmelsayed we havent added any restriction in the pods or namespace in which keda is deployed.Only Azure firewall is implemented in front of AKS with restriction in egress traffic and allowed the required rules for AKS to communicate outside.

For Keda is there any sepcific urls or ports to be allowed in Azure firewall

**Reply by tomkerkhove** [reply]
If this is AKS related, did you check this recommendation https://docs.microsoft.com/en-us/azure/aks/limit-egress-traffic#required-outbound-network-rules-and-fqdns-for-aks-clusters?

That's because this applies for master <-> worker node communication which is what is happening here. Master nodes will ask KEDA for metrics, but it should be allowed to do it. Also, check your NSGs.

That said, we should have it documented for sure so happy to improve that once it's solved.

**Reply by anoopmash** [reply]
@ahmelsayed @tomkerkhove Thanks for the input.

**Reply by nidiculageorge** [reply]
@tomkerkhove @ahmelsayed 

Thanks so much for the support. Issue has been resolved by adding a Network Rule in Azure firewall instead of Application Rule for API access.

(Source : AKS Subnet
Destination : AKS ClusterAPI Name
Port : 443)

Which removed the following error from the firewall


2/25/2021,   12:37:00.675 PM |   | N/A | N/A | Unknown Action | No rule matched | N/A | HTTPS request from   172.28.112.35:41380. Action: Deny. Reason: SNI TLS extension was missing.
-- | -- | -- | -- | -- | -- | -- | --

**Reply by tomkerkhove** [reply]
Happy to help!

**Reply by tomkerkhove** [reply]
I'll try to update the docs asap.

**Reply by ahmelsayed** [reply]
Thanks Tom :)
